{
  "candidate": "CVE-2009-2940",
  "description": "The pygresql module 3.8.1 and 4.0 for Python does not properly support the PQescapeStringConn function, which might allow remote attackers to leverage escaping issues involving multibyte character encodings.",
  "notes": {
    "jdstrand": "1:4.0-0ubuntu1 and higher has the fix affected versions have an escape_string() and escape_bytea() that uses PQescapeString() and PQescapeBytea() from PostgreSQL's libpq-fe.h. These are known to be problematic. The fix is to create pg_escape_string() and pg_escape_bytea() which use the safe PQescapeStringConn() and PQescapeByteaConn() functions, and then add them to the pgobj methods. Applications will have to be rewritten to use the new functions, specifically, something like this: cnx = pg.connect(...) ... escaped = pg.escape_string(str) to be: cnx = pg.connect(...) ... escaped = cnx.escape_string(str)"
  },
  "priority": "medium",
  "references": [
    "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2009-2940",
    "https://ubuntu.com/security/notices/USN-870-1"
  ],
  "assigned_to": "jdstrand",
  "public_date": "2009-10-22 16:30:00 UTC",
  "public_date_at_usn": "2009-10-22",
  "packages": {
    "dapper": {
      "pygresql": {
        "name": "pygresql",
        "status": "ignored",
        "note": "end of life",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    },
    "devel": {
      "pygresql": {
        "name": "pygresql",
        "status": "not-affected",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    },
    "hardy": {
      "pygresql": {
        "name": "pygresql",
        "status": "released",
        "note": "1:3.8.1-2ubuntu0.1",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    },
    "intrepid": {
      "pygresql": {
        "name": "pygresql",
        "status": "released",
        "note": "1:3.8.1-3ubuntu0.1",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    },
    "jaunty": {
      "pygresql": {
        "name": "pygresql",
        "status": "not-affected",
        "note": "1:4.0-0ubuntu1",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    },
    "karmic": {
      "pygresql": {
        "name": "pygresql",
        "status": "not-affected",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    },
    "upstream": {
      "pygresql": {
        "name": "pygresql",
        "status": "released",
        "note": "1:4.0-1",
        "patches": [
          {
            "source": "vendor",
            "text": "http://www.debian.org/security/2009/dsa-1911"
          }
        ]
      }
    }
  }
}
